helix-core/src/auto_pairs.rs

```diff
@@ -18,9 +20,98 @@ pub const PAIRS: &[(char, char)] = &[
     ('`', '`'),
 ];
 
-// [TODO] build this dynamically in language config. see #992
-const OPEN_BEFORE: &str = "([{'\":;,> \n\r\u{000B}\u{000C}\u{0085}\u{2028}\u{2029}";
-const CLOSE_BEFORE: &str = ")]}'\":;,> \n\r\u{000B}\u{000C}\u{0085}\u{2028}\u{2029}"; // includes space and newlines
+/// The type that represents the collection of auto pairs,
+/// keyed by the opener.
+#[derive(Debug, Clone)]
+pub struct AutoPairs(HashMap<char, Rc<AutoPair>>);
```

Comment:
Why `Rc<>` here? We generally use `Arc<>` but the `AutoPair` type seems simple enough to make `Copy`.

Reply:
Hmm, I made it Rc initially when there was a field that had potentially several `dyn Fn`s, that were for the same pair, but yeah, now that that's gone, this is probably fine to just copy. Will change.